src/crimson/os/seastore/transaction.h

```diff
@@ -68,16 +63,7 @@ class Transaction {
   void add_to_retired_set(CachedExtentRef ref) {
     ceph_assert(!is_weak());
     if (ref->is_initial_pending()) {
-      // We decide not to remove it from fresh_block_list because touching this
-      // will affect relative paddrs, and it should be rare to retire a fresh
-      // extent.
       ref->state = CachedExtent::extent_state_t::INVALID;
-      if (ref->is_inline()) {
-	write_set.erase(*ref);
-      } else {
-	// if ref is not relative, it must be in the delayed set
-	delayed_set.erase(*ref);
-      }
```

Comment:
I'm not quite following, don't we have to remove the extent from `Transaction::write_set` when it's retired at the state CachedExtent::extent_state_t::INITIAL_WRITE_PENDING? I thought extents in this state can be in the write_set.

Reply:
I don't see a problem with leaving it in the write_set, and it simplifies the rules.